internal/rootcoord/create_collection_task.go

```diff
@@ -259,10 +259,34 @@ func (t *createCollectionTask) validateSchema(schema *schemapb.CollectionSchema)
 	return validateFieldDataType(schema)
 }
 
-func (t *createCollectionTask) assignFieldID(schema *schemapb.CollectionSchema) {
-	for idx := range schema.GetFields() {
-		schema.Fields[idx].FieldID = int64(idx + StartOfUserFieldID)
+func (t *createCollectionTask) assignFieldID(schema *schemapb.CollectionSchema) error {
+	name2id := map[string]int64{}
+	for idx, field := range schema.GetFields() {
+		field.FieldID = int64(idx + StartOfUserFieldID)
+		name2id[field.GetName()] = field.GetFieldID()
+	}
+
+	for fidx, function := range schema.GetFunctions() {
+		function.InputFieldIds = make([]int64, len(function.InputFieldNames))
+		function.Id = int64(fidx)
```

Comment:
Let's do the same with fidx: `function.Id = int64(fidx) + StartOfUserFunctionId`

Reply:
We prepare add system function？Field id start at 100，because id less than 100 was system field.

Reply:
Not sure but let's leave room for that and keep user field/function id to have a consistent pattern